### PR TITLE
codeintel: Expose associated upload id on index records

### PR DESCRIPTION
**Pull request by efritz**
We expose `associated_index_id` on `lsif_uploads`. This adds the reverse relationship on the index model.

### CODE REVIEW

File: enterprise/internal/codeintel/stores/dbstore/indexes.go [hunk 1]
```diff
@@ -432,6 +443,7 @@ var indexColumnsWithNullRank = []*sqlf.Query{
 	sqlf.Sprintf(`u.execution_logs`),
 	sqlf.Sprintf("NULL"),
 	sqlf.Sprintf(`u.local_steps`),
+	sqlf.Sprintf(`` + indexAssociatedUploadIDQueryFragment + ``),
```
Comment (efritz) [inline enterprise/internal/codeintel/stores/dbstore/indexes.go:446]:
```suggestion
	sqlf.Sprintf(indexAssociatedUploadIDQueryFragment),
```

File: enterprise/internal/codeintel/stores/dbstore/indexes.go [hunk 2]
```diff
@@ -124,6 +126,12 @@ func (s *Store) GetIndexByID(ctx context.Context, id int) (_ Index, _ bool, err
 	return scanFirstIndex(s.Store.Query(ctx, sqlf.Sprintf(getIndexByIDQuery, id, authzConds)))
 }
 
+const indexAssociatedUploadIDQueryFragment = `
+(
+	SELECT MAX(id) FROM lsif_uploads WHERE associated_index_id = u.id
+) as associated_upload_id
```
Comment (eseliger) [inline enterprise/internal/codeintel/stores/dbstore/indexes.go:132]:
```suggestion
) AS associated_upload_id
```

**Review by eseliger — APPROVED** [review]

File: enterprise/internal/codeintel/stores/dbstore/indexes.go (hunk 1 above)
Comment (Strum355) [inline enterprise/internal/codeintel/stores/dbstore/indexes.go]: Is \`\` something special here?

Comment (efritz) [reply enterprise/internal/codeintel/stores/dbstore/indexes.go]: No that was a stupid find and replace - fixed it.

**Review by Strum355 — APPROVED** [review]

**Pull request merged**